File: src/dotnet/ArdsTools/Ards.NetCore.Adif.Api/Models/PropogationMode.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Ards.NetCore.Adif.Api.Models
{
    [Table("propogation_mode", Schema = "adif")]
    public partial class PropogationMode
    {
        [Column("id")]
        public int Id { get; set; }
        [Required]
        [Column("enumeration")]
        [StringLength(20)]
        public string Enumeration { get; set; }
        [Required]
        [Column("description")]
        [StringLength(120)]
        public string Description { get; set; }
    }
}

File: src/java/ards-adif-api/src/main/java/com/ards/adif/api/model/PropogationMode.java
package com.ards.adif.api.model;

import java.io.Serializable;
import javax.persistence.*;


/**
 * The persistent class for the propogation_mode database table.
 * 
 */
@Entity
@Table(name="propogation_mode")
@NamedQuery(name="PropogationMode.findAll", query="SELECT p FROM PropogationMode p")
public class PropogationMode implements Serializable {
	private static final long serialVersionUID = 1L;

	@Id
	@Column(unique=true, nullable=false)
	private Integer id;

	@Column(nullable=false, length=120)
	private String description;

	@Column(nullable=false, length=20)
	private String enumeration;

	public PropogationMode() {
	}

	public Integer getId() {
		return this.id;
	}

	public void setId(Integer id) {
		this.id = id;
	}

	public String getDescription() {
		return this.description;
	}

	public void setDescription(String description) {
		this.description = description;
	}

	public String getEnumeration() {
		return this.enumeration;
	}

	public void setEnumeration(String enumeration) {
		this.enumeration = enumeration;
	}

}